custom.d.ts

```diff
@@ -0,0 +1,5 @@
+declare namespace JSX {
+    interface IntrinsicElements {
+        'pwa-install': React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>
+    }
+}
```

Comment:
Q: why is this necessary? I have no idea - could we add a comment explaining why it's needed?

Reply:
It's adding definition for the pwa-install component into the JSX namespace so that typescript doesn't kill us.
```ts
export default function PWAInstaller({ ...props }) {
    return <pwa-install id={`pwa-install`} {...props}></pwa-install>
}
```
But i'll move it to the global.d.ts actually, no need for a new separate declaration file